Core/Src/Systems/External/Can/Can.c

```diff
@@ -0,0 +1,85 @@
+#include "../../../../Inc/Systems/External/Can/Can.h"
+#include "stdio.h"
+#include <string.h>
+#include <stdlib.h>
+
+int parseCanData(CanMessage* canMsg, const char* fn) {
+    FILE* file = fopen(fn, "r");
+    if (!file) {
+        perror("Failed to open CAN data file");
+        exit(EXIT_FAILURE);
+    }
+
+    char line[256];
+    int messageId = -1;
+
+    while (fgets(line, sizeof(line), file)) {
+        // Line contains the message ID (First Line)
+        if (strncmp(line, "BO_", 3) == 0) {
+            sscanf(line, "BO_ %d", &messageId);
+            canMsg->messageId = messageId;
+        }
+
+        // Line contains raw CAN data
+        if (strncmp(line, "Raw CAN Data", 3) == 0) {
```

Comment:
```"Raw CAN Data", 3``` compares the first three characters of the string ```"Raw CAN Data"``` - ```"Raw"``` to the line, 
so the line would still be considered valid CAN data ven if it only contain ```"Raw"``` instead of ```"Raw CAN Data"```.

It should be fine if there are no other data that starts with ```"Raw"```, but for more accurate comparison, the parameters should be:
```"Raw CAN Data", 12``` instead.